paasta_tools/cli/cmds/local_run.py

```diff
@@ -418,15 +419,17 @@ def docker_pull_image(docker_url):
     bindings due to the docker auth/registry transition. Once we are past Docker 1.6
     we can use better credential management, but for now this function assumes the
     user running the command has already been authorized for the registry"""
-    paasta_print("Please wait while the image (%s) is pulled..." % docker_url, file=sys.stderr)
+    paasta_print("Please wait while the image (%s) is pulled (times out after 30m)..." % docker_url, file=sys.stderr)
     DEVNULL = open(os.devnull, 'wb')
-    ret, output = _run('docker pull %s' % docker_url, stream=True, stdin=DEVNULL)
-    if ret != 0:
-        paasta_print(
-            "\nPull failed. Are you authorized to run docker commands?",
-            file=sys.stderr,
-        )
-        sys.exit(ret)
+    with open('/tmp/paasta-local-run-pull.lock', 'w') as f:
+        with timed_flock(f, seconds=1800):
```

Comment:
Would it be useful to do something other than just throw a `TimeoutError` exception here?

Reply:
A retry might make sense. Especially in the case of a Jenkins local-run for an itest, that would prevent the healthcheck from failing intermittently.

Reply:
Why? If we are going to retry, why not just wait longer?

If we would retry 10times, then I'll just make the timeout be 18000 instead

Reply:
You are right